modules/apps/object/object-service/src/main/java/com/liferay/object/internal/security/permission/resource/util/ObjectDefinitionResourcePermissionUtil.java

```diff
@@ -120,4 +136,76 @@ private static String _getPermissionsSupports(
 		return permissionsSupports;
 	}
 
+	private static String _getRootDescendantNodeObjectDefinitionsModelResource(
+			ObjectActionLocalService objectActionLocalService,
+			ObjectDefinitionPersistence objectDefinitionPersistence,
+			ObjectDefinition objectDefinition, TreeFactory treeFactory)
+		throws Exception {
+
+		if (!objectDefinition.isRootNode()) {
+			return StringPool.BLANK;
+		}
+
+		Tree tree = treeFactory.create(
+			objectDefinition.getObjectDefinitionId());
+
+		Iterator<Node> iterator = tree.iterator();
+
+		String modelResources = StringPool.BLANK;
+
+		String objectActionPermissionKeys = StringPool.BLANK;
+
+		while (iterator.hasNext()) {
+			Node node = iterator.next();
+
+			if (Objects.equals(
+					node.getObjectDefinitionId(),
+					objectDefinition.getObjectDefinitionId())) {
+
+				continue;
+			}
+
+			for (ObjectAction objectAction :
+					objectActionLocalService.getObjectActions(
+						node.getObjectDefinitionId(),
+						ObjectActionTriggerConstants.KEY_STANDALONE)) {
+
+				objectActionPermissionKeys = StringBundler.concat(
+					objectActionPermissionKeys, "\n\t\t\t<action-key>",
+					objectAction.getName(), "</action-key>");
+			}
+
+			if (StringUtil.equals(
+					objectActionPermissionKeys, StringPool.BLANK)) {
+
+				continue;
+			}
+
+			ObjectDefinition rootDescendantNodeObjectDefinition =
+				objectDefinitionPersistence.findByPrimaryKey(
+					node.getObjectDefinitionId());
+
+			String modelResourceHeader = StringBundler.concat(
+				"\n<model-resource>\n\t<model-name>",
```

Comment:
Do we really need these padding (\n and \t)?

Reply:
Actually I haven't try without these paddings, I will see if it works too

Reply:
It works!!